### ISSUE TITLE
Allow more sort-imports options

### ISSUE THREAD

**Issue by webOS101** [eslint/eslint#8470]
1. Eslint 3.19.0
2. More flexible import ordering rules
3. Allow grouping of alphabetized sections based on relativity to importing module.

Basically, our house rules alphabetize imports as follows:

```js
import Ramda from 'ramda';
import React from 'react';

import AnotherPeerComponent from '../AnotherPeerComponent';
import PeerComponent from '../PeerComponent';

import css from './Component.less';
import LocalComponent from './LocalComponent';
```

So, 'global' imports sorted, coming first.  Then relative modules  Then local level imports.

Would you be interested in a pull request that allowed this as an optional sort order?

**Comment by gyandeeps** [comment]
What you are looking for is already implemented in the plugin `eslint-plugin-imports` under [`order` rule](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md).

**Comment by webOS101** [comment]
So it is!  Looks very nice.  Still, the question stands about whether this is a useful addition to the core or if one should rely on this module.  I'm happy either way.

**Comment by platinumazure** [comment]
Hi @webOS101, Thanks for the issue.

We have a large number of core rules already, so we've set a pretty high bar for inclusion of new rules. Basically, except for really critical things, we [prefer that new rules be implemented in plugins](http://eslint.org/docs/developer-guide/contributing/new-rules). So since there is already a good import-related plugin out there, I would assume there would be little chance of such a rule being added to core.

ESLint plugins are discoverable using the npm package tag "eslintplugin", but it's pretty hard to search beyond that. I would be interested in seeing whether we could improve that experience somehow, either in npm or on our site. If you have any concrete suggestions along those lines, please let us know!

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
